src/components/esphome-button-menu.ts

```diff
@@ -0,0 +1,48 @@
+import "@material/mwc-menu";
+import type { Corner, Menu } from "@material/mwc-menu";
+import { css, html, LitElement, TemplateResult } from "lit";
+import { customElement, property, query } from "lit/decorators.js";
+
+@customElement("esphome-button-menu")
+export class ESPHomeButtonMenu extends LitElement {
+  @property() public corner: Corner = "TOP_START";
+
+  @query("mwc-menu", true) private _menu?: Menu;
+
+  public get items() {
+    return this._menu?.items;
+  }
+
+  public get selected() {
+    return this._menu?.selected;
+  }
+
+  protected render(): TemplateResult {
+    return html`
+      <div @click=${this._handleClick}>
+        <slot name="trigger"></slot>
+      </div>
+      <mwc-menu .corner=${this.corner}>
+        <slot></slot>
+      </mwc-menu>
+    `;
+  }
+
+  private _handleClick(): void {
```

Comment:
Whatever generates the click, (a link or a button), we should make sure to prevent any default behavior
```suggestion
  private _handleClick(ev): void {
    ev.preventDefault();
```